packages/core/README.md

```diff
@@ -4,45 +4,51 @@
 
 ## Usage
 
-#### Inputs/Outputs
-
-You can use this library to get inputs or set outputs:
+### Import the package
 
 ```js
+// javascript
 const core = require('@actions/core');
 
-const myInput = core.getInput('inputName', { required: true });
+// typescript
+import * as core from '@actions/core';
+```
 
-// Do stuff
+#### Inputs/Outputs
+
+Action inputs can be read with `getInput`.  Outputs be set with `setOutput` which makes them available to be mapped into inputs of other actions to ensure they are decoupled.
+
+```js
+const myInput = core.getInput('inputName', { required: true });
 
 core.setOutput('outputKey', 'outputVal');
 ```
 
 #### Exporting variables
 
-You can also export variables for future steps. Variables get set in the environment.
+Since steps run out of proc, you can use `exportVariable` to add it to this step and future steps environment blocks.
 
 ```js
-const core = require('@actions/core');
+core.exportVariable('envVar', 'Val');
+```
 
-// Do stuff
+Exporting a secret exports the variable but also registers the secret with the runner to ensure it is masked in logs.
 
-core.exportVariable('envVar', 'Val');
+```js
+core.exportSecret('myPassword', mypass);
 ```
 
 #### PATH Manipulation
 
-You can explicitly add items to the path for all remaining steps in a workflow:
+To make a tool's path available in the path for the remainder of the job (without altering the machine or containers state), use `addPath`.  The runner will prepend the path given to the jobs PATH.
 
 ```js
-const core = require('@actions/core');
-
-core.addPath('pathToTool');
+core.addPath('/path/to/mytool');
 ```
 
 #### Exit codes
 
-You should use this library to set the failing exit code for your action:
+You should use this library to set the failing exit code for your action.  If you status is set and the script runs to completion, that will lead to a success.
```

Comment:
This is confusing to me "If you status is set and the script runs to completion, that will lead to a success."

Could you possibly mean "If **no** status is set and the script runs to completion, that will lead to a success."